clang/include/clang/AST/APValue.h

```diff
@@ -249,13 +249,15 @@ class APValue {
   struct NoLValuePath {};
   struct UninitArray {};
   struct UninitStruct {};
+  struct ConstexprUnknown {};
 
   template <typename Impl> friend class clang::serialization::BasicReaderBase;
   friend class ASTImporter;
   friend class ASTNodeImporter;
 
 private:
   ValueKind Kind;
+  bool AllowConstexprUnknown = false;
```

Comment:
Would it make sense to use a bit-field here so we can pack this bit together with the `Kind`?

Reply:
Initializing a bit-field inline is a C++20 extension and it got messy when I tried to do the init manually but I can look at it again now that this is a much simpler set of code.

Reply:
Initializing a bit-field inline is a C++20 extension and so I don't think I can use it.

I set `AllowConstexprUnknown` everyplace I thought it mattered in `APValue` but I obtain different results. It means I am missing someplace where `APValue` is being initialized. I can try again and see if I can hunt down the places I missed but it left me a bit spooked that it was not obvious.